dask_ml/preprocessing/data.py

```diff
@@ -370,3 +370,231 @@ def transform(self, X, y=None):
+
+    def fit(self, X, y=None):
+        """Determine the categorical columns to be dummy encoded.
+
+        Parameters
+        ----------
+        X : pandas.DataFrame or dask.dataframe.DataFrame
+        y : ignored
+
+        Returns
+        -------
+        self
+        """
+        self.columns_ = X.columns
+        columns = self.columns
+        if columns is None:
+            columns = X.select_dtypes(include=['category']).columns
+        else:
+            for column in columns:
+                assert is_categorical_dtype(X[column]), "Must be categorical"
+
+        self.categorical_columns_ = columns
+        self.non_categorical_columns_ = X.columns.drop(
+            self.categorical_columns_)
+
+        if _HAS_CTD:
+            self.dtypes_ = {col: X[col].dtype
+                            for col in self.categorical_columns_}
+        else:
+            self.dtypes_ = {
+                col: (X[col].cat.categories, X[col].cat.ordered)
+                for col in self.categorical_columns_
+            }
+
+        left = len(self.non_categorical_columns_)
+        self.categorical_blocks_ = {}
+        for col in self.categorical_columns_:
+            right = left + len(X[col].cat.categories)
+            self.categorical_blocks_[col], left = slice(left, right), right
+
+        if isinstance(X, pd.DataFrame):
+            sample = X.iloc[:1]
+        else:
+            sample = X._meta_nonempty
+
+        self.transformed_columns_ = pd.get_dummies(
+            sample, drop_first=self.drop_first).columns
+        return self
+
+    def transform(self, X, y=None):
+        """Dummy encode the categorical columns in X
+
+        Parameters
+        ----------
+        X : pd.DataFrame or dd.DataFrame
+        y : ignored
+
+        Returns
+        -------
+        transformed : pd.DataFrame or dd.DataFrame
+            Same type as the input
+        """
+        # TODO: validation of columns, dtypes
+        if isinstance(X, pd.DataFrame):
+            return pd.get_dummies(X, drop_first=self.drop_first)
+        elif isinstance(X, dd.DataFrame):
+            return dd.get_dummies(X, drop_first=self.drop_first)
+        else:
+            raise TypeError
+
+    def inverse_transform(self, X):
+        """Inverse dummy-encode the columns in `X`
+
+        Parameters
+        ----------
+        X : array or dataframe
+            Either the NumPy, dask, or pandas version
+
+        Returns
+        -------
+        data : DataFrame
+            Dask array or dataframe will return a Dask DataFrame.
+            Numpy array or pandas dataframe will return a pandas DataFrame
+        """
+        if isinstance(X, np.ndarray):
+            X = pd.DataFrame(X, columns=self.transformed_columns_)
+
+        elif isinstance(X, da.Array):
+            # later on we concat(..., axis=1), which requires
+            # known divisions. Suboptimal, but I think unavoidable.
+            unknown = np.isnan(X.chunks[0]).any()
```

Comment:
@mrocklin, I'd be curious to hear your thoughts on how we can better handle unknown chunks/divisions here. The stuff here and further down (directly assigning `._chunks`) is pretty hacky.

In this case, since we're going `da.Array` with unknown chunks -> `dd.DataFrame`, we known we'll end up with a RangeIndex, so as long as we know the length of each chunk we can compute the divsions.

I *think* that having `dd.from_dask_array` took an optional `divisions` argument (or maybe `chunks`?) would be the cleanest way to support this. We could actually require that the argument be provided when there are any unknown divisions, since currently you end up with incorrect results when the dask array as unknown chunks.

Reply:
Do you need to know the number of rows in every chunk?  Why do you need accurate divisions here?

I agree that modifying `_chunks` feels odd.  We don't yet have a better mechanism.